### PR TITLE
Bump project version from 3.11 to 3.13

### PR DESCRIPTION
**Pull request by sheaf**
This PR bumps the Cabal version from 3.11 to 3.13, as the release branch for 3.12 has already been cut.

### CODE REVIEW

**Review by geekosaur — APPROVED** [review]
Looks okay to me, but CI thinks we're still depending on 3.11 somewhere.

**Comment by sheaf** [comment]
> Looks okay to me, but CI thinks we're still depending on 3.11 somewhere.

Yeah there's still some work to do, I don't know how these bumps were done in the past.

**Comment by Mikolaj** [comment]
@sheaf: here's the latest PR that does that (but it uses version 3.12 and this one needs 3.13): https://github.com/haskell/cabal/pull/9788 and here's the last PR that did this for master branch: https://github.com/haskell/cabal/pull/8844. Hope that helps.

**Comment by Mikolaj** [comment]
And it's also described in https://github.com/haskell/cabal/wiki/Making-a-release#c3-bumping-version-numbers

**Comment by sheaf** [comment]
I've updated the bootstrap plans and added a few missing version bumps. Let's see how far we get now.

**Comment by Mikolaj** [comment]
BTW, since this ticket was opened, the release checklist has changed and now also these steps need to be done on branch master (unless done already; probably a half is): https://github.com/haskell/cabal/issues/9729

**Comment by Mikolaj** [comment]
This is a part of the release process, so let's fast-track, unless anybody needs more time to review (I will now).

**Comment by Mikolaj** [comment]
@mergify rebase

**Review by geekosaur — APPROVED** [review]

File: cabal.project.bootstrap
Comment (ulysses4ever) [inline cabal.project.bootstrap:12]: Proliferation of `index-states` is concerning. Ideally, it'd be done with `import`'s.

Comment (geekosaur) [reply cabal.project.bootstrap:12]: I think #9901 makes that difficult currently.

Comment (ulysses4ever) [reply cabal.project.bootstrap:12]: Yeah, I haven't put enough thought into possible workarounds. I guess, this PR should be merged asap, and we can think about it more, later.

**Review by ulysses4ever — APPROVED** [review]

File: bootstrap/cabal-bootstrap-gen.cabal
Comment (Mikolaj) [inline bootstrap/cabal-bootstrap-gen.cabal:15]: The first release from these series will be exceptionally 3.12.0.0, but it's indeed unlikely anybody is bootstrapping until cabal-install is released in 3.12.1.0.

File: cabal.project.bootstrap
Comment (Mikolaj) [inline cabal.project.bootstrap:12]: Sigh. I will report about this duplicated index state and the breakage that necessitated this new file in https://github.com/haskell/cabal/issues/9901

**Review by Mikolaj — APPROVED** [review]
Thank you very much!

**Comment by Mikolaj** [comment]
I forgot it now merges after only a few jobs pass for an uknown reason. I will monitor that the rebased CI finishes.

**Comment by geekosaur** [comment]
Interestingly, there is nothing about this in the Mergify rulebase, so it's presumably a code change in Mergify. I wish I'd been able to find documentation on mergify.com, but it seems to all be marketing information. ☹️

**Comment by Mikolaj** [comment]
I suspect it's related to branch protection rules in Setting, but they do look fine as well.

**Comment by ulysses4ever** [comment]
> I forgot it now merges after only a few jobs pass for an uknown reason. I will monitor that the rebased CI finishes.

I missed when this problem appeared. I am reasonably sure that it's due to https://github.com/haskell/cabal/pull/9355. After it, Validate post job always "succeeds" instantaneously and can fail later on on the real runs, if there are any, but Mergify won't wait and once it sees "success" there, it merges right away. /cc @BinderDavid

**Comment by Mikolaj** [comment]
@geekosaur: I've bumped your repo permissions to Maintain. I wonder if you can see the Settings tab now.

**Comment by ulysses4ever** [comment]
> I wish I'd been able to find documentation on mergify.com

How is https://docs.mergify.com/?

**Comment by BinderDavid** [comment]
Should we open a separate issue for this? I fail to see who exactly is responsible here: mergify or the github branch protection rules. https://docs.mergify.com/workflow/actions/merge/ for example is quite unclear and vague:

> Mergify always respects the branch protection settings

but also 

> The following protection settings are only partially supported by Mergify:
Require status checks to pass before merging;

When I implemented the logic for the faster documentation CI I focused on getting the code correct for the branch protection logic that Github uses. I thought that mergify only provides additional convenience features on top of Github's rules, and that it can't override something that Github branch protection forbids, but now I am not so sure anymore.

**Comment by ulysses4ever** [comment]
@BinderDavid thanks for a prompt response! I think no one, including the approvers of your PR, could have anticipated this interaction between GH and Mergify (assuming this is the real reason of the problem, which is not a given). Indeed, opening a separate ticket would be good.

**Pull request merged**
